qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -254,6 +288,11 @@ def solve(
                 )
                 threshold_satisfied = True
                 break
+            if iteration == 1 and np.abs(max_grad[0]) < self._threshold:
+                raise QiskitNatureError(
+                    "Gradient choice is not suited as it leads to all non-zero gradients. "
+                    "Try a different gradient method."
+                )
```

Comment:
This is an attempt at catching bad gradient configurations. However, as it stands now it will never occur because line 284 will be executed beforehand.
You will need to move this section above that `if` statement or you can integrate the two like so:
```python
            if np.abs(max_grad[0]) < self._threshold:
                if iteration == 1:
                    raise QiskitNatureError(
                        "Gradient choice is not suited as it leads to all non-zero gradients. "
                        "Try a different gradient method."
                    )
                logger.info(
                    "Adaptive VQE terminated successfully with a final maximum gradient: %s",
                    str(np.abs(max_grad[0])),
                )
                threshold_satisfied = True
                break
```

Once you do this, the `test_natural_gradient` test case will no longer run into an awkward Python-internal error but instead will raise this warning.
You then have two options for that test case:
- figure out a natural gradient which works
- rewrite the test using `assertRaises` as done for example here: https://github.com/Qiskit/qiskit-nature/blob/main/test/converters/second_quantization/test_qubit_converter.py#L264